File: solutions/csharp/triangle/2/Triangle.cs
public static class Triangle
{
    private static bool ValidEdges(double a, double b, double c) =>
        a + b > c
        && b + c > a
        && a + c > b;

    public static bool IsScalene(double a, double b, double c) =>
        ValidEdges(a, b, c) && a != b && b != c && a != c;

    public static bool IsIsosceles(double a, double b, double c) =>
        ValidEdges(a, b, c) && !IsScalene(a, b, c);

    public static bool IsEquilateral(double a, double b, double c) =>
        ValidEdges(a, b, c) && a == b && b == c;
}

File: solutions/typescript/triangle/2/triangle.ts
export class Triangle {
  readonly sides: number[]
  readonly totalLength: number
  readonly isValid: boolean

  constructor(...sides: number[]) {
    this.sides = sides
    this.totalLength = sides.reduce((a, b) => a + b, 0)
    this.isValid = this.sides.every(x => x > 0 && this.totalLength - x >= x)
  }

  get isEquilateral(): boolean {
    return this.isValid && new Set(this.sides).size === 1
  }

  get isIsosceles(): boolean {
    return this.isValid && new Set(this.sides).size <= 2
  }

  get isScalene(): boolean {
    return this.isValid && new Set(this.sides).size === 3
  }
}
